pkg/components/queue_agent.go

```diff
@@ -316,12 +329,88 @@ func (qa *QueueAgent) prepareInitQueueAgentState() {
 		`fi`,
 	}
 
-	qa.initQAState.SetInitScript(strings.Join(script, "\n"))
-	job := qa.initQAState.Build()
+	qa.initQAStateJob.SetInitScript(strings.Join(script, "\n"))
+	job := qa.initQAStateJob.Build()
 	container := &job.Spec.Template.Spec.Containers[0]
 	container.EnvFrom = []corev1.EnvFromSource{qa.secret.GetEnvSource()}
 }
 
+func (qa *QueueAgent) needQAStateInit() bool {
+	return len(qa.tabletNodes) > 0
+}
+
+func (qa *QueueAgent) setConditionQAStatePreparedForUpdating(ctx context.Context) {
+	qa.ytsaurus.SetUpdateStatusCondition(ctx, metav1.Condition{
+		Type:    consts.ConditionQAStatePreparedForUpdating,
+		Status:  metav1.ConditionTrue,
+		Reason:  "QAStatePreparedForUpdating",
+		Message: "Queue agent state prepared for updating",
+	})
+}
+
+func (qa *QueueAgent) setConditionQAStateUpdated(ctx context.Context) {
+	qa.ytsaurus.SetUpdateStatusCondition(ctx, metav1.Condition{
+		Type:    consts.ConditionQAStateUpdated,
+		Status:  metav1.ConditionTrue,
+		Reason:  "QAStateUpdated",
+		Message: "Queue agent state updated",
+	})
+}
+
+func (qa *QueueAgent) initQAState(ctx context.Context, dry bool) (ComponentStatus, error) {
+	for _, tnd := range qa.tabletNodes {
+		tndStatus, err := tnd.Status(ctx)
+		if err != nil {
+			return tndStatus, err
+		}
+		if !IsRunningStatus(tndStatus.SyncStatus) {
+			// Wait for tablet nodes to proceed with operations archive init.
```

Comment:
```suggestion
			// Wait for tablet nodes to proceed with queue agent init.
```
This was a bit misleading copy-paste, I started to think how we can wait for op archive by checking tnds.